.ci/pytorch/common.sh

```diff
@@ -17,6 +17,8 @@ if [[ "${BUILD_ENVIRONMENT}" == *rocm* ]]; then
   export HSAKMT_DEBUG_LEVEL=4
   # improve rccl performance for distributed tests
   export HSA_FORCE_FINE_GRAIN_PCIE=1
+elif [[ "${BUILD_ENVIRONMENT}" == *xpu* ]]; then
+  export PYTORCH_TEST_WITH_XPU=1
```

Comment:
Looking a bit closer into this, could we just get rid of this environment variable altogether?  I mean `PYTORCH_TEST_WITH_XPU` and `TEST_WITH_XPU`.  IMO, mps provides a nice example https://github.com/pytorch/pytorch/blob/main/torch/testing/_internal/common_utils.py#L1247 and we can use something like `TEST_XPU = torch.backends.xpu.is_available()` without the need to manually set `PYTORCH_TEST_WITH_XPU` here

Reply:
Thanks @huydhn for the good suggestions, totally get your point. Yes, the `torch.backends.xpu.is_available()` will be ready in https://github.com/pytorch/pytorch/pull/116850. According to your suggestion, I have used `--xpu` to instead `PYTORCH_TEST_WITH_XPU` and `TEST_WITH_XPU`. Compared with `--mps`, we did below changes
- Reuse the `.ci/pytorch/test.sh` directly, Use `PYTHON_TEST_EXTRA_OPTION` to add `--xpu` in function `test_python` & `test_python_shard` when xpu ci test triggered.
- When the `--xpu` enabled, the `selected_tests` will be default test cases expected `XPU_BLOCKLIST`, else all xpu specific test in `XPU_TEST` list will not be scheduled. 

Please help to review again, thanks~